crates/biome_analyze/src/rule.rs

```diff
@@ -313,6 +315,80 @@ impl RuleSourceKind {
     }
 }
 
+/// Rule domains
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+#[cfg_attr(
+    feature = "serde",
+    derive(
+        schemars::JsonSchema,
+        serde::Deserialize,
+        serde::Serialize,
+        biome_deserialize_macros::Deserializable
+    )
+)]
+#[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
+pub enum RuleDomain {
+    /// React library rules
+    React,
+    /// Testing rules
+    Test,
+    /// Solid.js framework rules
+    Solid,
+    /// Next.js framework rules
+    Next,
+}
+
+impl RuleDomain {
+    /// If the project has one of these dependencies, the rule will be automatically applied, unless it's explicitly disabled by the configuration
+    ///
+    /// If the array is empty, it means that the rules that belong to a certain domain won't enable themselves automatically.
+    pub const fn manifest_dependencies(&self) -> &[&(&str, &str)] {
+        match self {
+            RuleDomain::React => &[&("react", ">=16.0.0")],
+            RuleDomain::Test => &[
+                &("jest", ">=26.0.0"),
+                &("mocha", ">=8.0.0"),
+                &("ava", ">=2.0.0"),
+                &("vitest", ">=1.0.0"),
+            ],
+            RuleDomain::Solid => &[&("solid", ">=1.0.0")],
+            RuleDomain::Next => &[&("react", ">=16.0.0"), &("next", ">=14.0.0")],
+        }
+    }
+
+    /// Global identifiers that should be added to the `globals` of the [crate::AnalyzerConfiguration] type
+    pub const fn globals(&self) -> &'static [&'static str] {
```

Comment:
Same as the previous comment.
```suggestion
    pub const fn globals(self) -> &'static [&'static str] {
```